src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -85,17 +86,19 @@ public class JnlpAgentEndpointResolver extends JnlpEndpointResolver {
     @NonNull
     private final List<String> jenkinsUrls;
 
-    private SSLSocketFactory sslSocketFactory;
+    private final String agentName;
 
     private String credentials;
 
     private String proxyCredentials;
 
     private String tunnel;
 
+    private SSLSocketFactory sslSocketFactory;
+
     private boolean disableHttpsCertValidation;
 
-    private final String agentName;
+    private HostnameVerifier hostnameVerifier;
```

Comment:
As elsewhere, putting mandatory user-specified arguments before optional ones, and putting user-specified arguments before computed things like SSL socket factories and hostname verifiers.